src/widget/input.rs

```diff
@@ -91,8 +104,8 @@ where
                         }
                     };
 
-                    // Clear message, we parsed it succesfully
-                    state.input = String::new();
+                    // Clear message and add it to history
+                    state.history.push_front(std::mem::take(&mut state.input));
```

Comment:
Can this grow unbounded?

Reply:
Good point, we should probably have some limit

Reply:
64905a2

I've truncated to 100 here. Shouldn't have much memory footprint and is more than enough.